android.toolchain.cmake

```diff
@@ -0,0 +1,27 @@
+
+if(ANDROID_NDK_TOOLCHAIN_INCLUDED)
+  return()
+endif(ANDROID_NDK_TOOLCHAIN_INCLUDED)
+include($ENV{ANDROID_NDK_HOME}/build/cmake/android.toolchain.cmake)
+
+# Debug and release flags.
+list(REMOVE_ITEM ANDROID_COMPILER_FLAGS -g)
+list(APPEND ANDROID_COMPILER_FLAGS_DEBUG -g)
+list(APPEND ANDROID_COMPILER_FLAGS_RELWITHDEBINFO -g ${ANDROID_COMPILER_FLAGS_RELEASE})
```

Comment:
It is not sufficient to change `ANDROID_COMPILER_FLAGS_*`. The values have already been copied into `CMAKE_C_FLAGS_*`, `CMAKE_CXX_FLAGS_*`,`CMAKE_ASM_FLAGS_*` further down in the included toolchain file and are not applied directly. So the replacements have to be applied to all of those variables:

```cmake
## Build without debug symbols in Release mode and add RelWithDebInfo mode
foreach(_var ANDROID_COMPILER_FLAGS CMAKE_C_FLAGS CMAKE_CXX_FLAGS CMAKE_ASM_FLAGS)
  string(REPLACE "-g " "" ${_var} "${${_var}}")
  set(${_var}_DEBUG "-g ${${_var}_DEBUG}")
  set(${_var}_RELWITHDEBINFO "-g ${${_var}_RELEASE}")
endforeach()
unset(_var)
```

Reply:
Thanks for the observation; I'll open a PR with the modification.